### PR TITLE
Set criteria options

### PR DESCRIPTION
**Pull request by shefalijoshi**
Author Checklist

- Changes address original issue? Yes
- Unit tests included and/or updated with changes? Yes
- Command line build passes? Yes
- Changes have been smoke-tested? Yes

### CODE REVIEW

File: src/plugins/condition/utils/evaluator.js [hunk 1]
```diff
@@ -1,7 +1,31 @@
...
+    for (let key in resultMap) {
+        if (resultMap.hasOwnProperty(key)) {
+            result = resultMap[key];
+            if (result) {
+                break;
+            }
+        }
+    }
+    return result;
 };
```
Comment (JoelMcKinnon) [inline src/plugins/condition/utils/evaluator.js]: Why not refactor these as computeCondition(trigger, resultMap)?

File: src/plugins/condition/utils/evaluator.js [hunk 2]
```diff
@@ -1,7 +1,31 @@
...
+    let result = false;
+    for (let key in resultMap) {
+        if (resultMap.hasOwnProperty(key)) {
+            result = resultMap[key];
+            if (!result) {
+                break;
+            }
+        }
+    }
+    return result;
```
Comment (psarram) [inline src/plugins/condition/utils/evaluator.js]: The code for `computeConditionForAll()` and `computeConditionForAny()` is identical except for the line that checks the `result`. Instead of repeating the same code, it would be better to define a common function and pass the trigger `any` or `all` depending on which of the 2 functions is calling it.

File: src/plugins/condition/components/ConditionEdit.vue [hunk 3]
```diff
@@ -262,18 +301,22 @@ export default {
             }
         },
         operationKeyChange(ev) {
-            if (ev.target.value !== 'isUndefined' && ev.target.value !== 'isDefined') {
```
Comment (psarram) [inline src/plugins/condition/components/ConditionEdit.vue]: A function is an action and so the name should indicate that.

File: src/plugins/condition/components/ConditionEdit.vue [hunk 4]
```diff
@@ -262,18 +301,22 @@ export default {
...
                 this.comparisonValueField = true;
             } else {
                 this.comparisonValueField = false;
             }
-            this.condition.definition.criteria[0].operation = this.selectedOperationKey;
-            this.persist();
+            this.updateConditionCriteria();
             //find the criterion being updated and set the operation property
         },
+        telemetryChange() {
```
Comment (psarram) [inline src/plugins/condition/components/ConditionEdit.vue]: A function is an action and so the name should indicate that.

File: src/plugins/condition/utils/evaluator.js (hunk 1 above)
Comment (shefalijoshi) [reply src/plugins/condition/utils/evaluator.js]: Mostly for clarity.

File: src/plugins/condition/utils/evaluator.js (hunk 2 above)
Comment (shefalijoshi) [reply src/plugins/condition/utils/evaluator.js]: I did this mostly for clarity than anything else as I mentioned above for Joel. I suppose since both of you noticed, I'll merge the 2.

Comment (shefalijoshi) [reply src/plugins/condition/utils/evaluator.js]: Done.

File: src/plugins/condition/utils/evaluator.js (hunk 1 above)
Comment (shefalijoshi) [reply src/plugins/condition/utils/evaluator.js]: Done

File: src/plugins/condition/components/ConditionEdit.vue (hunk 3 above)
Comment (shefalijoshi) [reply src/plugins/condition/components/ConditionEdit.vue]: Done

File: src/plugins/condition/components/ConditionEdit.vue (hunk 4 above)
Comment (shefalijoshi) [reply src/plugins/condition/components/ConditionEdit.vue]: Done

File: src/plugins/condition/components/ConditionEdit.vue [hunk 5]
```diff
@@ -259,20 +324,18 @@ export default {
                 this.condition.definition.output = this.selectedOutputKey;
             }
         },
-        operationKeyChange(ev) {
-            if (ev.target.value !== 'isUndefined' && ev.target.value !== 'isDefined') {
+        setInputValueVisibility(ev) {
+            if (this.selectedOperationKey !== 'isUndefined' && this.selectedOperationKey !== 'isDefined') {
```
Comment (psarram) [inline src/plugins/condition/components/ConditionEdit.vue]: This is hard to follow. Are you trying to check for the value not being undefined? You can use the typeOf operator like so `if (typeOf this.selectedOperationKey !== 'undefined')`

File: src/plugins/condition/components/ConditionCollection.vue [hunk 6]
```diff
@@ -113,6 +117,16 @@ export default {
...
+                let objId = this.openmct.objects.makeKeyString(obj.identifier);
+                let id = this.openmct.objects.makeKeyString(telemetryDomainObjectIdentifier);
+                return objId === id;
+            });
+            if (index > -1) {
+                this.telemetryObjs.splice(index, 1);
+            }
+        },
         addCondition(event, isDefault) {
             let conditionDO = this.getConditionDomainObject(!!isDefault);
```
Comment (psarram) [inline src/plugins/condition/components/ConditionCollection.vue]: DO in the variable name is not descriptive.

File: src/plugins/condition/components/ConditionEdit.vue [hunk 7]
```diff
@@ -239,17 +275,46 @@ export default {
...
+        },
         hasTelemetry() {
             return this.condition.definition.criteria.length && this.condition.definition.criteria[0].key;
         },
+        updateConditionCriteria() {
+            if (this.condition.definition.criteria.length) {
+                this.condition.definition.criteria[0].key = this.selectedTelemetryKey;
+                this.condition.definition.criteria[0].metaDataKey = this.selectedMetaDataKey;
+                this.condition.definition.criteria[0].operation = this.selectedOperationKey;
+                this.condition.definition.criteria[0].input = [this.operationValue];
```
Comment (psarram) [inline src/plugins/condition/components/ConditionEdit.vue]: Generally it's a good practice to define a variable for an object once if it's going to be accessed more than once. 
`let criteria = this.condition.definition.criteria[0];`

File: src/plugins/condition/components/ConditionCollection.vue [hunk 8]
```diff
@@ -35,6 +35,7 @@
             >
                 <div v-if="isEditing">
                     <ConditionEdit :condition-identifier="conditionIdentifier"
+                                   :telemetry="telemetryObjs"
                                    :current-condition-identifier="currentConditionIdentifier"
                                    @update-current-condition="updateCurrentCondition"
                                    @remove-condition="removeCondition"
```
Comment (psarram) [inline src/plugins/condition/components/ConditionCollection.vue]: Let's use camelCase for events and props to be consistent with the rest of the code base. We really need to update our coding style guide.

File: src/plugins/condition/components/ConditionEdit.vue (hunk 5 above)
Comment (shefalijoshi) [reply src/plugins/condition/components/ConditionEdit.vue]: No, it's comparing the 'name' property of the operation is 'isUndefined'. 
See this file: https://github.com/nasa/openmct/blob/set-criteria-options/src/plugins/condition/utils/operations.js#L159

File: src/plugins/condition/components/ConditionEdit.vue (hunk 7 above)
Comment (shefalijoshi) [reply src/plugins/condition/components/ConditionEdit.vue]: Yes agreed. This was going to change soon with the addCriteria work that Joel is working on so I left it as is. I'll change it.

File: src/plugins/condition/components/ConditionCollection.vue (hunk 8 above)
Comment (shefalijoshi) [reply src/plugins/condition/components/ConditionCollection.vue]: OK

Comment (shefalijoshi) [reply src/plugins/condition/components/ConditionCollection.vue]: If I change the attribute names to camelCase I see this linting error:
`ESLint: Attribute ':currentConditionIdentifier' must be hyphenated.(vue/attribute-hyphenation)`
Do you want me to still change it?

File: src/plugins/condition/components/ConditionEdit.vue (hunk 5 above)
Comment (psarram) [reply src/plugins/condition/components/ConditionEdit.vue]: In that case, i would expect the variable to be selectedOperationName.

Comment (shefalijoshi) [reply src/plugins/condition/components/ConditionEdit.vue]: Done.

File: src/plugins/condition/components/ConditionCollection.vue (hunk 6 above)
Comment (shefalijoshi) [reply src/plugins/condition/components/ConditionCollection.vue]: DONE

File: src/plugins/condition/components/ConditionEdit.vue (hunk 7 above)
Comment (shefalijoshi) [reply src/plugins/condition/components/ConditionEdit.vue]: Done

File: src/plugins/condition/components/ConditionCollection.vue (hunk 8 above)
Comment (JoelMcKinnon) [reply src/plugins/condition/components/ConditionCollection.vue]: VueJS docs say should be kebab-case for events. Pretty sure same for props and lint objects for those: https://vuejs.org/v2/guide/components-custom-events.html#Event-Names

**Comment by psarram** [comment]
@shefalijoshi The changes look good, thanks! The branch has conflicts thought. Please fix it so I can merge.

**Comment by psarram** [comment]
**Reviewer Checklist**

- Changes appear to address issue? Y
- Appropriate unit tests included? N
- Code style and in-line documentation are appropriate? Y
- Commit messages meet standards? Y
